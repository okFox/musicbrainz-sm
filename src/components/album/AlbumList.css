.Albums {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  width: 90vw;

  li {
    list-style: none;
    color: blue;
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    margin: 30px;
    max-width: 250px;
  }

  img {
    max-height: 250px;
    max-width: 230px;
  }
}