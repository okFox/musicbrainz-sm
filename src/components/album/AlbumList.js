import React from 'react';
import PropTypes from 'prop-types';
import AlbumItem from './AlbumItem';
import styles from './AlbumList.css';

const AlbumList = ({ albumArray, artistName }) => {

  const albumElements = albumArray.map((album, i) => (
    <li key={i}>
      <AlbumItem albumId={album.albumId} albumTitle={album.albumTitle} artistName={artistName} />
    </li>
  ));

  return (
    <ul className={styles.Albums}>
      {albumElements}
    </ul>
  );
};

AlbumList.propTypes = {
  albumArray: PropTypes.array.isRequired,
  artistName: PropTypes.string.isRequired
}

export default AlbumList;
