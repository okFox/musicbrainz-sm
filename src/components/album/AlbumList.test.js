import React from 'react';
import { shallow } from 'enzyme';
import AlbumList from './AlbumList';

describe('AlbumList component', () => {
  it('renders AlbumList', () => {
    const wrapper = shallow(<AlbumList albumArray={[{ albumId: '4', albumTitle: 'title' }]} artistName="name" />);
    expect(wrapper).toMatchSnapshot();
  });
});
