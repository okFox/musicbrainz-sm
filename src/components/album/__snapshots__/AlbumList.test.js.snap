// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumList component renders AlbumList 1`] = `
<ul
  className="Albums"
>
  <li
    key="0"
  >
    <AlbumItem
      albumId="4"
      albumTitle="title"
      artistName="name"
    />
  </li>
</ul>
`;
